stdlib/public/core/StringLegacy.swift

```diff
@@ -43,7 +43,7 @@ extension String {
       String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
         var total = 0
         for i in 0..<count {
-          let offset = i * repeatedUTF8.count
+          let offset = i &* repeatedUTF8.count
```

Comment:
`&*` isn't saturating multiplication and must not be used here. If you overflow then this will be very wrong, and in the context of memory-unsafe operations we don't want that at all.

Reply:
> If you overflow

This cannot overflow

Reply:
In the context (which it's been so long I'd forgotten), quite right! Thanks for reminding me; looks like I'd suggested this above. That said, this is *not* a saturating operation and one should not be misled about its general safety!

Reply:
Changed commit message.